app/src/main/java/com/poupa/vinylmusicplayer/misc/queue/StaticPlayingQueue.java

```diff
@@ -0,0 +1,458 @@
+                queue.get(i).index = i;
+                originalQueue.get(i).index = i;
+            }
+        }
+
+        if (from > currentPosition && to <= currentPosition) {
+            this.currentPosition = currentPosition + 1;
+        } else if (from < currentPosition && to >= currentPosition) {
+            this.currentPosition = currentPosition - 1;
+        } else if (from == currentPosition) {
+            this.currentPosition = to;
+        }
+
+        songsIsStale = true;
+    }
+
+    private int rePosition(int deletedPosition) {
+        int position = this.currentPosition;
+
+        if (deletedPosition < position) {
+            this.currentPosition = position - 1;
+        } else if (deletedPosition == position) { //the current position was deleted
+            if (queue.size() > deletedPosition) {
+                return position;
+            } else {
+                return position - 1;
+            }
+        }
+
+        return INVALID_POSITION;
+    }
+
+    // remove song at index position, numbering need to be redone for every song after this position (-1)
+    public int remove(int position) {
+        IndexedSong o = queue.remove(position);
+        originalQueue.remove(o.index);
+
+        for (int i = 0; i < queue.size(); i++) {
+            originalQueue.get(i).index = i;
+            if (queue.get(i).index >= o.index) {
+                queue.get(i).index = queue.get(i).index - 1;
+            }
+        }
+
+        songsIsStale = true;
+
+        return rePosition(position);
+    }
+
+    private int removeAllOccurrence(Song song) {
+        int hasPositionChanged = INVALID_POSITION;
+
+        for (int i = queue.size() - 1; i >= 0; i--) {
+            if (queue.get(i).id == song.id) {
+                int temp = remove(i);
+                if (temp != INVALID_POSITION) {
+                    hasPositionChanged = temp;
+                }
+            }
+        }
+
+        songsIsStale = true;
+
+        return hasPositionChanged;
+    }
+
+    public int removeSongs(@NonNull List<Song> songs) {
+        int hasPositionChanged = INVALID_POSITION;
+
+        for (Song song : songs) {
+            int temp = removeAllOccurrence(song);
+            if (temp != INVALID_POSITION) {
+                hasPositionChanged = temp;
+            }
+        }
+
+        return hasPositionChanged;
+    }
+
+    public void clear() {
+        songs.clear();
+        queue.clear();
+        originalQueue.clear();
+    }
+
+    private void revert() {
+        queue = new ArrayList<>(originalQueue);
+        songsIsStale = true;
+    }
+
+    /* -------------------- queue getter info -------------------- */
+
+    public boolean openQueue(@Nullable final ArrayList<Song> playingQueue, final int startPosition, final boolean startPlaying, int shuffleMode) {
+        if (playingQueue != null && !playingQueue.isEmpty() && startPosition >= 0 && startPosition < playingQueue.size()) {
```

Comment:
maybe invert this logic to return false early, then do this otherwise? reduces indentation